controllers/managedocs_controller_test.go

```diff
@@ -676,6 +687,26 @@ var _ = Describe("ManagedOCS controller", func() {
 					By("Creating a prometheus resource")
 					utils.WaitForResource(k8sClient, ctx, promTemplate.DeepCopy(), timeout, interval)
 
+					By("Creating an alertmanager resource")
+					utils.WaitForResource(k8sClient, ctx, amTemplate.DeepCopy(), timeout, interval)
+				})
+				It("should create reconciled resources", func() {
+					if testReconciler.DeploymentType != providerDeploymentType {
+						Skip(fmt.Sprintf("Skipping the test as it is not required by %v", testReconciler.DeploymentType))
+					}
+					// Create a valid add-on parameters secret
+					secret := addonParamsSecretTemplate.DeepCopy()
+					secret.Data["size"] = []byte("1")
+					secret.Data["enable-mcg"] = []byte("false")
+					secret.Data["onboarding-validation-key"] = []byte("test-validation-key")
+					Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+					By("Creating a storagecluster resource")
+					utils.WaitForResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
```

Comment:
We have different deployment types now so we need to validate the content as well. 
I think a deep equal on a desired outcome would be good